### ISSUE TITLE
Video embeds not working Discord

### ISSUE THREAD

**Issue by AEtherPi** [Wikidepia/InstaFix#67]
It only shows the preview image, even when enlarged.
![image](https://github.com/Wikidepia/InstaFix/assets/59917665/cb9413d5-cd2b-4d85-ad38-0ee85bc24464)

**Comment by danvayn** [comment]
Might be specifically a reels issue, I have the same issue with my reel link.

**Comment by shelbyKiraM** [comment]
I know youtube-dl has been having issues getting reel URLs for me a couple/few weeks? I can tight-click, copy URL from vid in Safari on macOS using the [Baking Soda](https://apps.apple.com/us/app/baking-soda-tube-cleaner/id1601151613) plugin tho… Lé sigh. Meta tryin' to be smart is making us NORMAL NERDS sadge.

**Comment by stanleylaw** [comment]
i noticed this one doesn't even load up.

[](https://www.instagram.com/reel/C14fSnbL9Ax/)

**Comment by GeorgeWL** [comment]
I only just started seeing this issue, was fine earlier in the week.

Ah example link
https://www.ddinstagram.com/reel/C3-9hPNIUzS/?igsh=

**Comment by barraIhsan** [comment]
It's all now working guys, the link given by the OP is also worked

**Comment by rnnc** [comment]
doesnt work, the following video isn't loading

https://www.instagram.com/p/C1cKtgZtWaV/

**Comment by NicoTheCinderace** [comment]
> It's all now working guys, the link given by the OP is also worked

Nope.

**Comment by Wikidepia** [comment]
InstaFix get the video URL from Instagram embed, sometimes they are blocked by Instagram, for example:

- https://www.instagram.com/p/C1cKtgZtWaV/embed/captioned, when the post is blocked you cant play this link from yout browser, and only shows its thumbnail and "Watch on Instagram"

There is no bypass for this except to create a dummy account and scrape from it, but it will be easily blocked by IG and hard to maintain

**Comment by barraIhsan** [comment]
> InstaFix get the video URL from Instagram embed, sometimes they are blocked by Instagram, for example:
> 
>     * https://www.instagram.com/p/C1cKtgZtWaV/embed/captioned, when the post is blocked you cant play this link from your browser, and only shows its thumbnail and "Watch on Instagram"
> 
> 
> There is no bypass for this except to create a dummy account and scrape from it, but it will be easily blocked by IG and hard to maintain

hmm this one right here, 
![image](https://github.com/Wikidepia/InstaFix/assets/57800056/c02f7582-0bec-43e4-9301-0232e5e3d6c8)
is showing the video on discord embed, but if we open [the embed caption on the browser](https://www.instagram.com/reel/C3-9hPNIUzS/embed/captioned), it shows "Watch on Instagram" too

**Comment by Wikidepia** [comment]
> is showing the video on discord embed, but if we open [the embed caption on the browser](https://www.instagram.com/reel/C3-9hPNIUzS/embed/captioned), it shows "Watch on Instagram" too

There is other method used by InstaFix, sometimes IG does not block scraping directly from their graphql API (https://github.com/Wikidepia/InstaFix/blob/87e06bd9bfcf7b0df4760b8ad7b4eb6743684dfe/handlers/data/data.go#L325). Scraping in general is very fragile and IG change their anti bot stuff

**Comment by barraIhsan** [comment]
yeah, https:/ddinstagram.com/reel/C50Yr1MRtXh/ this one didn't embed the video
![Screenshot_2024-04-18-19-39-57-819_com.discord.png](https://github.com/Wikidepia/InstaFix/assets/57800056/c071488d-13f6-4635-a708-25258c317680)

(because I want to share it with my friend) I tried downloading it with [cobalt.tools](https://cobalt.tools) and it also didn't work
![Screenshot_2024-04-18-19-43-56-355_org.mozilla.firefox_beta.png](https://github.com/Wikidepia/InstaFix/assets/57800056/fa49f627-202e-4fbe-808d-04b1e9387b64)

but using [yt-dlp](https://github.com/yt-dlp/yt-dlp) works fine tho....

**Comment by barraIhsan** [comment]
yeah, https:/ddinstagram.com/reel/C50Yr1MRtXh/ this one didn't embed the video
![Screenshot_2024-04-18-19-39-57-819_com.discord.png](https://github.com/Wikidepia/InstaFix/assets/57800056/c071488d-13f6-4635-a708-25258c317680)

(because I want to share it with my friend) I tried downloading it with [cobalt.tools](https://cobalt.tools) and it also didn't work
![Screenshot_2024-04-18-19-43-56-355_org.mozilla.firefox_beta.png](https://github.com/Wikidepia/InstaFix/assets/57800056/fa49f627-202e-4fbe-808d-04b1e9387b64)

but using [yt-dlp](https://github.com/yt-dlp/yt-dlp) works fine tho....

**Comment by barraIhsan** [comment]
It seems that most of the links that were "broken" before, will fix themselves if you wait long enough.

for example, the reels I sent 3d ago are now embedding properly
![image](https://github.com/Wikidepia/InstaFix/assets/57800056/b8d7e029-003c-418d-b91b-dc4e37424ba5)

**Comment by GeorgeWL** [comment]
https://github.com/Wikidepia/InstaFix/issues/82#issuecomment-2142535246
